src/repositories/user.repository.test.ts

```diff
@@ -0,0 +1,21 @@
+import { Container } from 'typedi';
+import { Connection, getCustomRepository, getRepository } from 'typeorm';
+import { Database } from '../libs/database';
+import { UserRepository } from '../repositories/user.repository';
+
+describe('user repository test', async () => {
+
+  const database = Container.get(Database);
+
+  before(async () => {
+    await database.connect();
+  });
+  after(async () => {
+    await database.disconnect();
+  });
+
+  it('get all users', async () => {
+    console.log(await getCustomRepository(UserRepository).getUsers());
```

Comment:
Why use the `getCustomRepository`? I haven't tried but we can do the following instead.

`let users = await server.get('/users/').expect(200).then((resp) => resp.body);`

Should we not let it go pass through the endpoint?

Reply:
I see, reason I did not use the endpoint is that because I test only the custom repository itself

Reply:
Do we have to test the custom repository? Shouldn't it be in the controller instead?

Reply:
I guess we should really test that, because they are not the same as controller. I mean we applied a SRP for the repository and should have a test itself, relying on controller test may somehow broaden the tests, and will be hard to maintain,
For example if repository tests are failing it we all know that it is on the repository folder bug
But if we test only the controller, we have two options controller and repository because controller also used the repository. 

having a test on individual responsibility makes easier to pinpoint the failing test
though it may be hard for the developer to create a test on repository but it's for them to create or not but for me I think it is necessary